numpy/core/include/numpy/experimental_dtype_api.h

```diff
@@ -321,6 +327,28 @@ typedef int (PyArrayMethod_StridedLoop)(PyArrayMethod_Context *context,
         NpyAuxData *transferdata);
 
 
+/**
+ * Query an ArrayMethod for the initial value for use in reduction.
+ *
+ * @param context The arraymethod context, mainly to access the descriptors.
+ * @param reduction_is_empty Whether the reduction is empty, when it is the
+ *     default value is required, otherwise an identity value to start the
+ *     the reduction.  These might differ, examples:
+ *     - `0.0` as default for `sum([])`.  But `-0.0` would be the correct
+ *       identity as it preserves the sign for `sum([-0.0])`.
+ *     - We use no identity for object, but `0` and `1` for sum and prod.
+ *     - `-inf` or `INT_MIN` for `max` is an identity, but at least `INT_MIN`
+ *       not a good *default* when there are no items.
```

Comment:
I had to squint at this a little bit to understand what was going on, so I'd suggest the following reword to make things clearer:

```suggestion
 * @param reduction_is_empty Whether the reduction is empty. When it is, the
 *     default value for the identity might differ, for example:
 *     - `0.0` is the default for `sum([])`.  But `-0.0` is the correct
 *       identity otherwise as it preserves the sign for `sum([-0.0])`.
 *     - We use no identity for object, but `0` and `1` for sum and prod.
 *     - `-inf` or `INT_MIN` for `max` is an identity, but at least `INT_MIN`
 *       not a good *default* when there are no items.
```

If any of my edits change the meaning of what you were going for please discard, but maybe try re-wording because I'm still confused about the second example at least. Also this documentation is copy/pasted in `array_method.h`, so make sure to keep that in sync.

Reply:
Thanks, the object case should be: [...] , but use the defaults `0` and `1` for sum and prod" with no elements.  (will change)